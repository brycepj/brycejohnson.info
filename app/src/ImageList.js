import React from "react";
import styled from "styled-components";

import { bryceAndEmmy, rachaelAndEmmy, hikeImage } from "./assets/index";

const Container = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  align-content: center;
`;

const Image = styled.img`
  max-height: 300px;
  margin: 5px;
`;

function ImageList() {
  return (
    <Container>
      <Image src={bryceAndEmmy} alt="Bryce and Emmy" />
      <Image src={rachaelAndEmmy} alt="Rachael and Emmy" />
      <Image src={hikeImage} alt="Rachael, Bryce and Emmy on a hike" />
    </Container>
  );
}

export default ImageList;
