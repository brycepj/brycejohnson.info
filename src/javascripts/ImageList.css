.image-list {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  align-content: center;
}

img {
  max-height: 300px;
  margin: 5px;
}

