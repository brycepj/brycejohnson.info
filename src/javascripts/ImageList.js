const React = require('react');
const css = require('./ImageList.css');

module.exports = function ImageList() {
  return (
    <div className='image-list'>
      <img src="/assets/bne-min.jpeg" alt="Bryce and Emmy"/>
      <img src="/assets/rne-min.jpeg" alt="Rachael and Emmy"/>
      <img src="/assets/hike-min.jpg" alt="Rachael, Bryce and Emmy on a hike"/>
    </div>
  );
}
