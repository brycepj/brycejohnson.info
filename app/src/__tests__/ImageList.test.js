import React from "react";
import { shallow } from "enzyme";
import ImageList from "../ImageList";

describe("ImageList", () => {
  it("renders properly", () => {
    const result = shallow(<ImageList />);
    expect(result).toMatchSnapshot();
  });
});
