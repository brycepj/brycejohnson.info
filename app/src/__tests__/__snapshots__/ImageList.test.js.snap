// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.img
          alt="Bryce and Emmy"
          src="bne-min.jpeg"
/>,
        <styled.img
          alt="Rachael and Emmy"
          src="rne-min.jpeg"
/>,
        <styled.img
          alt="Rachael, Bryce and Emmy on a hike"
          src="hike-min.jpg"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "Bryce and Emmy",
          "src": "bne-min.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "Rachael and Emmy",
          "src": "rne-min.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "Rachael, Bryce and Emmy on a hike",
          "src": "hike-min.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.img
            alt="Bryce and Emmy"
            src="bne-min.jpeg"
/>,
          <styled.img
            alt="Rachael and Emmy"
            src="rne-min.jpeg"
/>,
          <styled.img
            alt="Rachael, Bryce and Emmy on a hike"
            src="hike-min.jpg"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "Bryce and Emmy",
            "src": "bne-min.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "Rachael and Emmy",
            "src": "rne-min.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "Rachael, Bryce and Emmy on a hike",
            "src": "hike-min.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
